apollo-converters/pojo/src/main/java/com/apollographql/android/converter/pojo/ApolloConverterFactory.java

```diff
@@ -21,11 +27,26 @@
 public final class ApolloConverterFactory extends Converter.Factory {
   private final Map<Type, ResponseFieldMapper> responseFieldMappers;
   private final Map<ScalarType, CustomTypeAdapter> customTypeAdapters;
+  private final Moshi moshi;
```

Comment:
Moshi lib is tiny: 457 methods only, we can eventually remove it and generate own serialization (the same as we did with ResponseReader), but for now to save time lets leave.

As for Gson, we could allow user to define own lib for serialization request but then user must  handle custom scalar type serialization in request by himself. So in this case he is going to have 2 places where define scalar type conversion: first for response (in `com.apollographql.android.converter.pojo.CustomTypeAdapter`), second for Gson.

In current implementation he defines these conversion in one place for both request / response: 
```
CustomTypeAdapter<Date> dateCustomTypeAdapter = new CustomTypeAdapter<Date>() {
      @Override public Date decode(String value) {
        try {
          return DATE_FORMAT.parse(value);
        } catch (ParseException e) {
          throw new RuntimeException(e);
        }
      }

      @Override public String encode(Date value) {
        return DATE_FORMAT.format(value);
      }
    };
....
new ApolloConverterFactory.Builder()
    .withCustomTypeAdapter(CustomType.DATETIME, dateCustomTypeAdapter)
```